Makefile

```diff
@@ -6,19 +6,51 @@ all: jsrun
 LLVM_CONFIG ?= llvm-config
 #CFLAGS+=-O1
 #CFLAGS+=-O3 -DNDEBUG
+#CFLAGS+=-O2
 #CXXFLAGS+=-O0 -g
 #CFLAGS+=-Werror
+CFLAGS+=-DDUK_OPT_NO_ASSERTIONS=1
 CFLAGS+=-DDUK_OPT_UNDERSCORE_SETJMP=1
 CFLAGS+=-DDUK_OPT_FORCE_ALIGN=32
 CFLAGS+=-DDUK_OPT_DEBUG=3
 #CFLAGS+=-DDUK_OPT_DDDPRINT=1 -DDUK_OPT_DDPRINT=1 -DDUK_OPT_DPRINT=1
 
-CC=~/sdk/bin/clang
-CFLAGS+=-msoft-float -mabi=sandbox -cheri-linker
+CHERI_SDK?=~/sdk
+
+VERSION?=x86
+
+.if $(VERSION) == cheri128 || $(VERSION) == cheri256
+CC=$(CHERI_SDK)/bin/cheri-unknown-freebsd-clang
+CFLAGS+=-msoft-float
+CFLAGS+=-mabi=sandbox -cheri-linker
+.if $(VERSION) == cheri128
+CFLAGS+=-mllvm -cheri128
+.endif
+#CFLAGS+=-mllvm -cheri-no-global-bounds
+CFLAGS+=-DDUK_USE_PACKED_TVAL=1
+CFLAGS+=--sysroot=$(CHERI_SDK)/sysroot
+CFLAGS+=-I/usr/include/edit
+CFLAGS+=-g
+LDFLAGS+=-mabi=sandbox -cheri-linker
```

Comment:
It should never be needed.  You can add -no-cheri-linker if you want the old behaviour (but then things won't work with the standard CheriBSD libraries)